psutil/tests/test_contracts.py

```diff
@@ -173,8 +173,10 @@ def test_rlimit(self):
         self.assertEqual(hasattr(psutil.Process, "rlimit"), LINUX or FREEBSD)
 
     def test_io_counters(self):
+        not_on_linux = (LINUX and not os.path.exists('/proc/1/io'))
```

Comment:
what's the meaning of this?

Reply:
It's Linux, but the thing doesn't exist here. I'm open for any better name.

Reply:
I see. It's OK, let's let it fail.

Reply:
It will constantly fail e.g. on my sparc machine because this simply does not exist there, annoying me during the stabilizations for Gentoo because the testsuite is marked as failed.

Reply:
These tests are supposed to be "alarms" testing API availability depending on the OS. I prefer to have them failing.

Reply:
But the API is not generally available on Linux. That's all what I'm testing here: if these iocounters are available. Thinking about that this test could result in a false negative for the (unlikely) case that one has locked down /proc access more than usual, e.g. to not show processes of other users.

If you don't like this one just cherry-pick the first one.

Reply:
> But the API is not generally available on Linux

It is generally available, at least on the majority of Linux i686 / x86 archs / distros I've seen. Most likely it's not available on Sparc, in which case I prefer to know with an explicit failure. 
I understand it is annoying to see a failure on your  specific arch / distro, but (in my mind) these tests checking API availability assume psutil is running on the most common arch / distro, not the "uncommon one", and are designed to (fail) be like this on purpose.

> If you don't like this one just cherry-pick the first one.

Wouldn't know how to do that.

Reply:
This is a kernel configuration thing. We have even bug reports that [this is failing on amd64](https://bugs.gentoo.org/701962) ([and again](https://bugs.gentoo.org/710646)).